sw/host/tests/manuf/manuf_cp_test_lock/src/main.rs

```diff
@@ -0,0 +1,155 @@
+const TEST_UNLOCK_TOKEN_PREIMAGE: [u32; 4] = [0xAAAAAAAA, 0xAAAAAAAA, 0xAAAAAAAA, 0xAAAAAAAA];
+/// Post-image of the TEST_UNLOCK token as generated by `gen_test_unlock_token.py`.
+const TEST_UNLOCK_TOKEN_POSTIMAGE: [u32; 4] = [0xe6b9ec3f, 0x67fd75f3, 0x618ae065, 0xc0bb8f81];
+
+fn main() -> anyhow::Result<()> {
+    let opts = Opts::from_args();
+    opts.init.init_logging();
+
+    let transport = opts.init.init_target()?;
+
+    execute_test!(provision_token, &opts, &transport);
+    execute_test!(test_lock, &opts, &transport);
+    execute_test!(test_unlock, &opts, &transport);
+
+    Ok(())
+}
+
+/// Provision the `TEST_UNLOCK` token, programming it into OTP.
+fn provision_token(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let jtag = reset_to_tap(opts, transport, JtagTap::RiscvTap)?;
+
+    OtpParam::write_param(
+        &*jtag,
+        DamParam::TestUnlockToken,
+        &TEST_UNLOCK_TOKEN_POSTIMAGE,
+    )
+    .context("failed to write TEST_UNLOCK token to OTP")?;
+
+    // Read back the token to verify it was written.
+    let mut test_unlock_token = [0xff; 4];
+    OtpParam::read_param(&*jtag, DamParam::TestUnlockToken, &mut test_unlock_token)
+        .context("failed to read back TEST_UNLOCK token from OTP")?;
+
+    assert_eq!(
+        test_unlock_token, TEST_UNLOCK_TOKEN_POSTIMAGE,
+        "check correct TEST_UNLOCK token can be read back from OTP"
+    );
+
+    jtag.disconnect()?;
+
+    Ok(())
+}
+
+/// Test that the device can be locked.
+fn test_lock(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let jtag = reset_to_tap(opts, transport, JtagTap::LcTap)?;
+
+    // Check that LC state starts in `TEST_UNLOCKED0`.
+    let state = jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?;
+    assert_eq!(state, DifLcCtrlState::TestUnlocked0.redundant_encoding());
+
+    lc_transition::trigger_lc_transition(
+        transport,
+        jtag.clone(),
+        DifLcCtrlState::TestLocked0,
+        None,
+        true,
+        opts.init.bootstrap.options.reset_delay,
+    )
+    .context("failed to trigger transition to TEST_UNLOCKED0")?;
+
+    // Check that LC state has transitioned to `TEST_LOCKED0`.
+    let state = jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?;
+    assert_eq!(state, DifLcCtrlState::TestLocked0.redundant_encoding());
+
+    jtag.disconnect()?;
+
+    Ok(())
+}
+
+/// Test that the device can be unlock using the provisioned token.
+fn test_unlock(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let jtag = reset_to_tap(opts, transport, JtagTap::LcTap)?;
+
+    // Check that LC state is currently `TEST_LOCKED0`.
+    let state = jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?;
+    assert_eq!(state, DifLcCtrlState::TestLocked0.redundant_encoding());
+
+    lc_transition::trigger_lc_transition(
+        transport,
+        jtag.clone(),
+        DifLcCtrlState::TestUnlocked0,
```

Comment:
If you start in TEST_UNLOCKED0, [you can only transition to TEST_LOCKED0](https://opentitan.org/book/doc/security/specs/device_life_cycle/index.html#architecture). I think you want to transition to TEST_UNLOCKED1 here.

Reply:
Ah, this explains all those LC states. Thanks.